libraries/AP_HAL_SITL/sitl_gps.cpp

```diff
@@ -1202,7 +1202,7 @@ void SITL_State::_update_gps(double latitude, double longitude, float altitude,
     d.speedN = speedN;
     d.speedE = speedE;
     d.speedD = speedD;
-    d.have_lock = have_lock;
+    d.have_lock = !_sitl->gps_disable;
```

Comment:
Pointing out that this is a behavior change, GPS will immediately start with lock.

Reply:
That was old behavior, no?

Reply:
I see that you and @tridge do not like the SITL change. I can remove it if you really think it's not useful.
Actually I think it's, I couldn't otherwise simulate AUTO_SWITCH properly.

Reply:
> That was old behavior, no?

No, this method is called with `false` initially, which would make `d.have_lock` be false. That won't happen with your change.

> I see that you and @tridge do not like the SITL change. I can remove it if you really think it's not useful.

Nobody said the change wasn't good. I just pointed out a behavior change and @tridge asked about why a new parameter was needed.